types/mydecimal.go

```diff
@@ -1212,6 +1229,30 @@ func (d *MyDecimal) ToBin(precision, frac int) ([]byte, error) {
 	return bin, err
 }
 
+/*
+ToHashKey removes leading zeros and trailing zeros and generates a key;
+Two keys generated by different precision or fraction can be compared with memcmp directly;
+
+For example:
+		1.1		decimal(2, 1)
+		1.10	decimal(3, 2)
+		01.100	decimal(5, 3)
+	these numbers above will generate a same key.
+*/
+func (d *MyDecimal) ToHashKey() ([]byte, error) {
+	_, digitsInt := d.removeLeadingZeros()
+	_, digitsFrac := d.removeTrailingZeros()
+	prec := digitsInt + digitsFrac
+	if prec == 0 { // zeroDecimal
+		prec = 1
+	}
+	buf, err := d.ToBin(prec, digitsFrac)
+	if err == ErrTruncated {
+		err = nil // trailing zeros are truncated, we ignore this error
```

Comment:
Is this error caused by the shorter `digitsFrac`?

Reply:
Our digitsFrac returned by `removeTrailingZeros()` includes all digits after the point, but the trailing zeros;
So if we got this error, these truncated digits must be the trailing zeros, and we just ignore them;
I will add some comments;